CMakeLists.txt

```diff
@@ -33,6 +34,7 @@ file(GLOB_RECURSE SRC_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp")
 
 add_library(${PROJECT_NAME} ${SRC_FILES})
 add_library(up-cpp::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}  -fprofile-arcs -ftest-coverage")
```

Comment:
Should this be controlled with a build flag? We probably don't want to be producing coverage reports in production builds.

The gcovr documentation [recommends adding a "PROFILE" build type](https://www.gcovr.com/en/stable/cookbook.html#out-of-source-builds-with-cmake) to cmake for this (although "COVERAGE" might make more sense).

It looks like a `--coverage` flag was added to gcc as the new standard way to build and link for gcovr (this was news to me 😅 ). It's supposed in gcc 11.4 (our minimum supported version), so switching to that would be preferred.

Reply:
This doesn't produce the ut report, this just prepares the output files needed to make such a report.
I tried the --coverage flag and it didn't work here.  I wanted to use it, but it produces an error, hence leaving what worked alone.

Reply:
The `--coverage` flag is not the only topic discussed here. At a minimum a new build type needs to be added to prevent release builds from including all the cruft for generating coverage data.

Reply:
Done with "Coverage" build type